.github/workflows/test.yml

```diff
@@ -1,323 +1,85 @@
-                  ]
-                  END
-                  )
-                  EXAMPLE_DIRECTORIES="${EXAMPLE_DIRECTORIES//'%'/'%25'}"
-                  EXAMPLE_DIRECTORIES="${EXAMPLE_DIRECTORIES//$'\n'/'%0A'}"
-                  EXAMPLE_DIRECTORIES="${EXAMPLE_DIRECTORIES//$'\r'/'%0D'}"
-                  echo "::set-output name=example_directories::$EXAMPLE_DIRECTORIES"
-            - id: should_run_tests
+                  directories: |
+                      ./examples
+                      ./tests
+                  exclude_dirs: |
+                      ${{ env.IS_FEATURE == 'true' && format('
+                         ./examples/basic_bitcoin
+                         ./examples/bitcoin_psbt
+                         ./examples/ckbtc
+                         ./tests/end_to_end/http_server/ethers_base
+                         ./tests/end_to_end/http_server/http_outcall_fetch
+                         ./tests/end_to_end/http_server/ic_evm_rpc
+                         ./tests/property/candid_rpc/class_api/stable_b_tree_map
+                         ./tests/property/candid_rpc/functional_api/stable_b_tree_map
+                         ') || '' }}
+
+    determine-should-run-tests:
+        name: Determine if tests should run
+        runs-on: ubuntu-latest
+        outputs:
+            should_run_tests: ${{ steps.determine_should_run_tests.outputs.should_run_tests }}
+        steps:
+            - uses: actions/checkout@v4
+
+            - id: determine_should_run_tests
+              # Determine if tests should run for this pr. If it's a manually submitted release candidate no tests will run.
               run: |
                   BRANCH_NAME="${{ github.head_ref }}"
                   RELEASE_VERSION="${BRANCH_NAME:9}"
                   COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")
-                  if [[ "${{ contains(github.head_ref, 'release--') }}" == "true" && "$COMMIT_MESSAGE" != "azle-bot automated release $RELEASE_VERSION" ]]
-                  then
-                    ./publish-github-action.sh $RELEASE_VERSION ${{ toJSON(steps.example_directories.outputs.example_directories) }}
+
+                  IS_AUTOMATED_RELEASE="false"
+                  if [[ "$COMMIT_MESSAGE" == "azle-bot automated release $RELEASE_VERSION" ]]; then
+                      IS_AUTOMATED_RELEASE="true"
+                  fi
+
+                  if [[ "${{env.IS_RELEASE_CANDIDATE}}" == "true" && "$IS_AUTOMATED_RELEASE" == "false" ]]; then
+                      echo "should_run_tests=false" >> "$GITHUB_OUTPUT"
                   else
-                    echo "::set-output name=should_run_tests::true"
+                      echo "should_run_tests=true" >> "$GITHUB_OUTPUT"
                   fi
-    basic-integration-tests:
-        needs: release-candidate-deploy
+
+    run-test:
+        # Tests can either run against the current code of Azle found in the repository, or the code deployed by the GitHub Action to npm.
```

Comment:
Can you redo the comments everywhere? Seems like maybe these could be updated, and we should use our variables etc to explain things better

Reply:
it's gone I think the other comments are sufficient